File: node/src/domain/Produto.ts
import Dimensao from "./Dimensao"
import Peso from "./Peso";

export default class Produto{

    constructor(readonly id: string, readonly description: string, readonly valor: number, readonly dimensao?: Dimensao, readonly peso?: Peso){}

    public getIndiceCalculoFrete() :number{
        const volume = this.dimensao ? this.dimensao.getVolume(): 0;
        const densidade = this.peso ? this.peso.calcularDensidade(volume) : 0
        return volume * (densidade / 100)
    }
}

File: golang/src/domain/produto.go
package domain

type Produto struct {
	Codigo    string
	Descricao string
	Valor     float64
	Dimensao  Dimensao
	Peso      Peso
}

func (produto Produto) getIndiceCalculoFrete() float64 {
	volume := produto.Dimensao.GetVolume()
	densidade := produto.Peso.CalcularDensidade(volume)

	return volume * (densidade / 100)
}
